packages/fast-element/src/directives/repeat.ts

```diff
@@ -0,0 +1,378 @@
+
+        for (let i = 0; i < newLen; ++i) {
+            view = views[i] = factory.create(true)!;
+            view.bind(items![i]);
+        }
+    }
+
+    private createAndBindNewViewsByKey(indexMap: IndexMap): void {
+        let view: ISyntheticView;
+        const factory = this.directive.template;
+        const views = this.views;
+        const mapLen = indexMap.length;
+        const items = this.items;
+
+        for (let i = 0; i < mapLen; ++i) {
+            if (indexMap[i] === -2) {
+                view = factory.create(true)!;
+                view.bind(items![i]);
+                views.splice(i, 0, view);
+            }
+        }
+
+        if (views.length !== mapLen) {
+            throw new Error(`viewsLen=${views.length}, mapLen=${mapLen}`);
+        }
+    }
+
+    private attachViews(indexMap: IndexMap | undefined): void {
+        let view: ISyntheticView;
+
+        const views = this.views;
+        const location = this.location;
+
+        if (indexMap === void 0) {
+            for (let i = 0, ii = views.length; i < ii; ++i) {
+                view = views[i];
+                view.insertBefore(location);
+            }
+        } else {
+            for (let i = 0, ii = views.length; i < ii; ++i) {
+                if (indexMap[i] !== i) {
+                    view = views[i];
+                    view.insertBefore(location);
+                }
+            }
+        }
+    }
+
+    private attachViewsKeyed(): void {
+        let view: ISyntheticView;
+        const { views, location } = this;
+        for (let i = 0, ii = views.length; i < ii; ++i) {
+            view = views[i];
+            view.insertBefore(location);
+        }
+    }
+
+    private sortViewsByKey(oldLength: number, indexMap: IndexMap): void {
+        // TODO: integrate with tasks
+        const location = this.location;
+        const views = this.views;
+        const newLen = indexMap.length;
+        synchronizeIndices(views, indexMap);
+
+        // this algorithm retrieves the indices of the longest increasing subsequence of items in the repeater
+        // the items on those indices are not moved; this minimizes the number of DOM operations that need to be performed
+        const seq = longestIncreasingSubsequence(indexMap);
+        const seqLen = seq.length;
+
+        let next: ISyntheticView;
+        let j = seqLen - 1;
+        let i = newLen - 1;
+        let view: ISyntheticView;
+        for (; i >= 0; --i) {
+            view = views[i];
+            if (indexMap[i] === -2) {
+                view.insertBefore(location);
+            } else if (j < 0 || seqLen === 1 || i !== seq[j]) {
+                // view.attach();
+            } else {
+                --j;
+            }
+
+            next = views[i + 1];
+            if (next !== void 0) {
+                // view.nodes!.link(next.nodes);
+            } else {
+                // view.nodes!.link(location);
+            }
```

Comment:
If we aren't performing work in these do we need to execute them? Could we just comment out the block for now?

Reply:
See comment below.